core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPreAggregateCaseAggregations.java

```diff
@@ -424,6 +424,19 @@ public void testDoesNotFireForNonCaseAggregation()
                         "GROUP BY col_varchar");
     }
 
+    @Test
+    public void testInfiniteOptimizeLoop()
```

Comment:
In the future, it won't be clear why this is called "testInfiniteOptimizeLoop". Rename it to `testIssue22666()`